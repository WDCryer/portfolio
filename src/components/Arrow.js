import React, { memo } from "react";
import styles from "./Arrow.module.css";

const Arrow = ({ direction = "right", size = "3px", className = "" }) => (
  <i
    className={`${styles.arrow} ${styles[direction]} ${className}`}
    style={{
      borderWidth: `0 ${size} ${size} 0`,
      padding: size
    }}
  />
);

export default memo(Arrow);
