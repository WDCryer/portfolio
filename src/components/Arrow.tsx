import React, { memo, ReactElement } from "react";

import styles from "./Arrow.module.css";

export type Direction = "right" | "left" | "up" | "down";
interface Props {
  readonly direction: Direction;
  readonly className?: string;
}

const Arrow = ({
  direction = "right",
  className = ""
}: Props): ReactElement => (
  <i className={`${styles.arrow} ${styles[direction]} ${className}`} />
);

export default memo(Arrow);
