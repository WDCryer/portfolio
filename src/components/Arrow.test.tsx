import React from "react";
import Arrow, { Direction } from "./Arrow";

describe("Arrow component", () => {
  const directions: Direction[] = ["right", "left", "up", "down"];

  test.each(directions)(
    "should render properly with direction %s",
    (direction): void => {
      expect(<Arrow direction={direction} />).toMatchSnapshot();
    }
  );
});
