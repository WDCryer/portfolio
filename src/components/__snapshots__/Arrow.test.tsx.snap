// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arrow component should render properly with direction down 1`] = `
<Memo(Arrow)
  direction="down"
/>
`;

exports[`Arrow component should render properly with direction left 1`] = `
<Memo(Arrow)
  direction="left"
/>
`;

exports[`Arrow component should render properly with direction right 1`] = `
<Memo(Arrow)
  direction="right"
/>
`;

exports[`Arrow component should render properly with direction up 1`] = `
<Memo(Arrow)
  direction="up"
/>
`;
